### PR TITLE
Packet decoder + DB parser (moved from pipboyrelay/#5)

### PR DESCRIPTION
**Pull request by AlexanderDzhoganov**
I have done some clean-up on the database parser. Haven't integrated my changes with the already existing code in `parser.js` as I haven't read up on the Dissolve library. Maybe somebody would be willing to further improve on this, but I think it works OK for now.

### CODE REVIEW

**Comment by rgbkrk** [comment]
> Maybe somebody would be willing to further improve on this, but I think it works OK for now.

We can go with this and iterate.

I'd rather not use buffertools though because of it being a native extension, though I can't tell which functionality you're using from first glance.

File: lib/pipdecode.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          this.emit('game_busy');
+          break;
+        case Channels.DatabaseUpdate:
+          this.emit('db_update', data);
+          break;
+        case Channels.LocalMapUpdate:
+          this.emit('localmap_update', data);
+          break;
+        case Channels.CommandResponse:
+          this.emit('command_response', JSOn.parse(data.toString()));
```
Comment (rgbkrk) [inline lib/pipdecode.js]: Missing a capital N here in `JSON.parse`

**Comment by rgbkrk** [comment]
I'm going to merge this and address a few things in a follow up.

Also adding you as a maintainer.

**Comment by AlexanderDzhoganov** [comment]
Thanks for adding me as a maintainer. I have addressed the `buffertools` issue in [this PR](https://github.com/rgbkrk/pipboylib/pull/11).

File: lib/pipdecode.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+  emitter.on('data', function(data) {
+    this.buffer = this.buffer.concat(data);
+
+    if(!this.expectedSize) {
+      this.expectedSize = this.buffer.readUInt32LE(0) + 5;
+    }
+
+    if(this.buffer.length >= this.expectedSize) {
+      var size = this.buffer.readUInt32LE(0);
```
Comment (rgbkrk) [inline lib/pipdecode.js]: I just noticed this is defined here but not used. Should we verify that `size` still `=== this.expectedSize`?

**Comment by rgbkrk** [comment]
In case anyone is wondering I favor a model of [Optimistic Merging](http://hintjens.com/blog:106) and of shared maintenance. If you contribute, you have a share. :smile:

**Pull request merged**